### PR TITLE
:bug: Update known_types to add x-kubernetes-int-or-string for intStr

### LINKED ISSUE #304: IntOrString is not structural in generated OpenAPIv3 CRD
I'm using kubebuilder v2.0.0-beta.0 with a api Go type that includes `v1.PodSpec`.
The generated CRD yaml can be installed by fails to be structural with condition errors like:

```
        Forbidden: must be empty to be structural, spec.validation.openAPIV3Schema.properties[spec].properties[custom].items.properties[podSpec].properties[containers].items.properties[lifecycle].properties[postStart].properties[httpGet].properties[port].type:
        Required value: must not be empty for specified object fields, spec.validation.openAPIV3Schema.properties[spec].properties[custom].items.properties[podSpec].properties[containers].items.properties[lifecycle].properties[postStart].properties[tcpSocket].properties[port].anyOf[0].type:
        Forbidden: must be empty to be structural, spec.validation.openAPIV3Schema.properties[spec].properties[custom].items.properties[podSpec].properties[containers].items.properties[lifecycle].properties[postStart].properties[tcpSocket].properties[port].anyOf[1].type:
        Forbidden: must be empty to be structural, spec.validation.openAPIV3Schema.properties[spec].properties[custom].items.properties[podSpec].properties[containers].items.properties[lifecycle].properties[postStart].properties[tcpSocket].properties[port].type:
        Required value: must not be empty for specified object fields, spec.validation.openAPIV3Schema.properties[spec].properties[custom].items.properties[podSpec].properties[containers].items.properties[lifecycle].properties[preStop].properties[httpGet].properties[port].anyOf[0].type:
        Forbidden: must be empty to be structural, spec.validation.openAPIV3Schema.properties[spec].properties[custom].items.properties[
```

I would expect the auto-generated OpenAPIV3 definitions to be structural.

### PR DESCRIPTION
**Pull request by tamalsaha**
Signed-off-by: Tamal Saha <tamal@appscode.com>

Fixes #304

This is an alternative for https://github.com/kubernetes-sigs/controller-tools/pull/312 .

### CODE REVIEW

File: pkg/crd/known_types.go [hunk 1]
```diff
@@ -77,10 +77,7 @@ var KnownPackages = map[string]PackageOverride{
 
 	"k8s.io/apimachinery/pkg/util/intstr": func(p *Parser, pkg *loader.Package) {
 		p.Schemata[TypeIdent{Name: "IntOrString", Package: pkg}] = apiext.JSONSchemaProps{
-			AnyOf: []apiext.JSONSchemaProps{
-				{Type: "string"},
-				{Type: "integer"},
-			},
+			XIntOrString: true,
```
Comment (sttts) [inline pkg/crd/known_types.go]: the anyOf could be left here for backwards compatibility (in addition to the extension).

Comment (tamalsaha) [reply pkg/crd/known_types.go]: Fixed!

File: pkg/crd/known_types.go [hunk 2]
```diff
@@ -77,6 +77,7 @@ var KnownPackages = map[string]PackageOverride{
 
 	"k8s.io/apimachinery/pkg/util/intstr": func(p *Parser, pkg *loader.Package) {
 		p.Schemata[TypeIdent{Name: "IntOrString", Package: pkg}] = apiext.JSONSchemaProps{
+			XIntOrString: true,
```
Comment (sttts) [inline pkg/crd/known_types.go]: little, but important detail: order must be integer, string, compare https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190425-structural-openapi.md#formal-definition

Comment (tamalsaha) [reply pkg/crd/known_types.go]: :open_mouth: Fixed!

**Pull request merged**